utils/wasm-builder/src/wasm_project.rs

```diff
@@ -220,15 +215,69 @@ impl WasmProject {
             let wasm_meta_path = self
                 .original_dir
                 .join([file_base_name, ".meta.txt"].concat());
-            let wasm_meta_hash_path = self.original_dir.join(".metahash");
 
             smart_fs::write_metadata(wasm_meta_path, metadata)
                 .context("unable to write `*.meta.txt`")?;
 
-            smart_fs::write(wasm_meta_hash_path, format!("{:?}", metadata.hash()))
-                .context("unable to write `.metahash`")?;
+            source_code = format!(
+                r#"{source_code}
+#[allow(improper_ctypes)]
```

Comment:
the only thing I consider here is to simplify this mod. can't see the reason for so many lines of code on purpose

Reply:
We can avoid type aliases and some consts indeed.

Reply:
repr(c) type may be also skipped because we don't assert result further. just [u8; 36] const array filled with zeroes to keep it on stack and throw ptr safely